From 12ecfcbe78f93301be9eafe9e03dba90638623fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Neum=C3=A4rker?= <xdch47@posteo.de>
Date: Fri, 22 Nov 2019 17:47:25 +0100
Subject: [PATCH] intial logformat: "<DATETIME> [<LEVEL>]: <DESCR>"

        set `LstdFlags` to `Ldate | Ltime | Llevelinitial` as this is
        more straightforward for the user and simpler to match for
        intrusion prevention software as fail2ban or sshguard.
---
 modules/log/conn_test.go    | 8 ++++----
 modules/log/console_test.go | 2 +-
 modules/log/file_test.go    | 6 +++---
 modules/log/flags.go        | 2 +-
 modules/log/smtp_test.go    | 2 +-
 modules/log/writer_test.go  | 2 +-
 6 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/modules/log/conn_test.go b/modules/log/conn_test.go
index cc3d758fa9..9dd42f843e 100644
--- a/modules/log/conn_test.go
+++ b/modules/log/conn_test.go
@@ -40,7 +40,7 @@ func TestConnLogger(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 
 	logger := NewConn()
 	connLogger := logger.(*ConnLogger)
@@ -112,7 +112,7 @@ func TestConnLoggerCloseBeforeSend(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 
 	logger := NewConn()
 
@@ -126,7 +126,7 @@ func TestConnLoggerFailConnect(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 
 	logger := NewConn()
 
@@ -169,7 +169,7 @@ func TestConnLoggerClose(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 
 	logger := NewConn()
 	connLogger := logger.(*ConnLogger)
diff --git a/modules/log/console_test.go b/modules/log/console_test.go
index a028b5b875..e3f2389ee9 100644
--- a/modules/log/console_test.go
+++ b/modules/log/console_test.go
@@ -64,7 +64,7 @@ func TestConsoleLogger(t *testing.T) {
 	}
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 
 	cw := NewConsoleLogger()
 	realCW := cw.(*ConsoleLogger)
diff --git a/modules/log/file_test.go b/modules/log/file_test.go
index 38279315ab..b434125679 100644
--- a/modules/log/file_test.go
+++ b/modules/log/file_test.go
@@ -24,7 +24,7 @@ func TestFileLoggerFails(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 	//filename := filepath.Join(tmpDir, "test.log")
 
 	fileLogger := NewFileLogger()
@@ -52,7 +52,7 @@ func TestFileLogger(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 	filename := filepath.Join(tmpDir, "test.log")
 
 	fileLogger := NewFileLogger()
@@ -155,7 +155,7 @@ func TestCompressFileLogger(t *testing.T) {
 
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 	filename := filepath.Join(tmpDir, "test.log")
 
 	fileLogger := NewFileLogger()
diff --git a/modules/log/flags.go b/modules/log/flags.go
index 992fc62ddb..5578a1b6bf 100644
--- a/modules/log/flags.go
+++ b/modules/log/flags.go
@@ -31,7 +31,7 @@ const (
 	Lmedfile = Lshortfile | Llongfile
 
 	// LstdFlags is the initial value for the standard logger
-	LstdFlags = Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial
+	LstdFlags = Ldate | Ltime | Llevelinitial
 )
 
 var flagFromString = map[string]int{
diff --git a/modules/log/smtp_test.go b/modules/log/smtp_test.go
index 216d555215..afbd20b542 100644
--- a/modules/log/smtp_test.go
+++ b/modules/log/smtp_test.go
@@ -17,7 +17,7 @@ import (
 func TestSMTPLogger(t *testing.T) {
 	prefix := "TestPrefix "
 	level := INFO
-	flags := LstdFlags | LUTC | Lfuncname
+	flags := Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC | Lfuncname
 	username := "testuser"
 	password := "testpassword"
 	host := "testhost"
diff --git a/modules/log/writer_test.go b/modules/log/writer_test.go
index 886dd58fb3..df2f6e6980 100644
--- a/modules/log/writer_test.go
+++ b/modules/log/writer_test.go
@@ -41,7 +41,7 @@ func TestBaseLogger(t *testing.T) {
 	b := WriterLogger{
 		out:    c,
 		Level:  INFO,
-		Flags:  LstdFlags | LUTC,
+		Flags:  Ldate | Ltime | Lmedfile | Lshortfuncname | Llevelinitial | LUTC,
 		Prefix: prefix,
 	}
 	location, _ := time.LoadLocation("EST")
